### PR TITLE
rgw/sfs: standardize log levels

### LINKED ISSUE #781: Standardize sfs log levels
Let's standardize sfs log levels. This came up during the devs call 2023-10-25.

**Describe the solution you'd like** 

Define our own log subsystem to be able to adjust logging independent of the rest of RGW.

Define constants to group ceph log levels to something more human understandable.

From the devs meeting we roughly like to have:

- ERROR (-1)
   - Something unrecoverable. Must always go to log irregardless of log level set by user.
   - Database corruption, ceph_abort
- Startup Message (0 or 1)
- Important, Warning (0)
- Info (1)
- Verbose (10)
  - GC ran
  - we did a thing and found it weird enough to log
- Debug (15)
  - entered a function, left function
  - we wrote stuff to disk
- Trace (> 20)

**Describe alternatives you've considered**

**Additional context**

Running benchmarks with --debug-rgw 10 instead of --debug-rgw 0 takes time:
11h -> 17h

### PR DESCRIPTION
**Pull request by jecluis**
Fixes: aquarist-labs/s3gw#781

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/rgw/driver/sfs/sfs_log.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#define lsfs_info_for(_dpp, _whom) lsfs_dout_for(_dpp, SFS_LOG_INFO, _whom)
+
+#define lsfs_verb(_dpp)            lsfs_dout(_dpp, SFS_LOG_VERBOSE)
+#define lsfs_verb_for(_dpp, _whom) lsfs_dout_for(_dpp, SFS_LOG_VERBOSE, _whom)
+
+#define lsfs_debug(_dpp)            lsfs_dout(_dpp, SFS_LOG_DEBUG)
+#define lsfs_debug_for(_dpp, _whom) lsfs_dout_for(_dpp, SFS_LOG_DEBUG, _whom)
+
+#define lsfs_trace(_dpp)            lsfs_dout(_dpp, SFS_LOG_TRACE)
+#define lsfs_trace_for(_dpp, _whom) lsfs_dout_for(_dpp, SFS_LOG_TRACE, _whom)
```
Comment (irq0) [inline src/rgw/driver/sfs/sfs_log.h:79]: ❤️

File: src/rgw/driver/sfs/sfs_log.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+//
+//  MEGA_TRACE: TRACE on steroids.
+//
+#define SFS_LOG_ERROR      -1
+#define SFS_LOG_IMPORTANT  0
+#define SFS_LOG_INFO       1
+#define SFS_LOG_VERBOSE    10
+#define SFS_LOG_DEBUG      15
+#define SFS_LOG_TRACE      20
+#define SFS_LOG_MEGA_TRACE 30
```
Comment (irq0) [inline src/rgw/driver/sfs/sfs_log.h:49]: I like the name, but can't think of a use for something above trace.

**Comment by irq0** [comment]
I like it!

**Review by irq0 — APPROVED** [review]
LGTM

Optional: Remove SFS_LOG_MEGA_TRACE, because no use case at the moment

**Comment by jecluis** [comment]
> Optional: Remove SFS_LOG_MEGA_TRACE, because no use case at the moment

I like it too much, and I'm sure someone will at some point use it.

**Review by 0xavi0 — APPROVED** [review]
lg2m

**Pull request merged**